### ISSUE TITLE
An option to send prompts to stderr (or the platform equivalent)

### ISSUE THREAD

**Issue by gschjetne** [ajalt/clikt#21]
Is it possible to add an `err` option to `TermUi.prompt()` like `TermUi.echo()`? That would be useful for tools where the user is inclined to pipe stdout somewhere and is not really interesting in preserving password prompts and the like.

**Comment by ajalt** [comment]
If I understand you correctly, it sounds like you want to throw away most Clikt output, but still have interactive prompt.

I don't think that makes sense semantically, but Clikt is a pragmatic library, so if it's useful in practice, I would consider adding it.

Unfortunatley, it's not possible to implement you use case. The only mechanism the JVM provides for detecting whether you have an interactive console is to check if `System.console()` is null, and it is null if any stream is redirected. So if you pipe stdout or stdin somewhere, the JVM acts like stderr is also redirected. So you wouldn't be able to interact with it anyway.

**Comment by gschjetne** [comment]
Interesting, I didn't know that. I suppose it might be a portability tradeoff.

[I did some looking around](https://superuser.com/questions/569432/why-can-i-see-password-prompts-through-redirecting-output) to see how other programs would prompt for passwords. Seems like programs like SSH  [access the TTY directly](https://github.com/openssh/openssh-portable/blob/90e51d672711c19a36573be1785caf35019ae7a8/openbsd-compat/readpassphrase.c) instead, bypassing any redirection. But I suppose that would be well out of the scope of Clikt.

But thanks for hearing me out, I learned something today. And of course, thanks for the library!
